samples/simple-chat/csharp/FunctionApp/Functions.cs

```diff
@@ -3,184 +3,89 @@
+            var req = (HttpRequest)executingContext.Arguments["req"];
+            var connectionInfo = (SignalRConnectionInfo)executingContext.Arguments["connectionInfo"];
 
-            return Convert.ToBase64String(Encoding.UTF8.GetBytes(source));
-        }
+            // get header
+            var myHeader = req.Headers["myheader"];
 
-        private static string GetBase64DecodedString(string source)
-        {
-            if (string.IsNullOrEmpty(source))
+            // validate token
+            var result = TokenProvider.ValidateToken(req);
+            
+            if (result.Status == AccessTokenStatus.Valid)
             {
-                return source;
-            }
-
-            return Encoding.UTF8.GetString(Convert.FromBase64String(source));
-        }
-
+                // resolve the identity
+                string identity = result.Principal.Identity.Name;
 
-        public static class EventGridTriggerCSharp
-        {
-            [FunctionName("onConnection")]
-            public static Task EventGridTest([EventGridTrigger]EventGridEvent eventGridEvent,
-                [SignalR(HubName = "simplechat")]IAsyncCollector<SignalRMessage> signalRMessages)
+                // override connectionInfo argument
+                connectionInfo.AccessToken = serviceManager.GenerateClientAccessToken(Constants.HubName, identity, new List<Claim> {new Claim("myheader", myHeader)});
+            }
+            else
             {
-                if (eventGridEvent.EventType == "Microsoft.SignalRService.ClientConnectionConnected")
-                {
-                    var message = ((JObject)eventGridEvent.Data).ToObject<SignalREvent>();
-
-                    return signalRMessages.AddAsync(
-                        new SignalRMessage
-                        {
-                            ConnectionId = message.ConnectionId,
-                            Target = "newConnection",
-                            Arguments = new[] { new ChatMessage
-                            {
-                                // ConnectionId is not recommand to send to client directly.
-                                // Here's a simple encryption for an easier sample.
-                                ConnectionId = GetBase64EncodedString(message.ConnectionId),
-                            }}
-                        });
-                }
-
-                return Task.CompletedTask;
+                connectionInfo.AccessToken = "Error while validating negotiate function token"; // todo: return with detailed error message
             }
-        }
 
-        public class ChatMessage
-        {
-            public string Sender { get; set; }
-            public string Text { get; set; }
-            public string Groupname { get; set; }
-            public string Recipient { get; set; }
-            public string ConnectionId { get; set; }
-            public bool IsPrivate { get; set; }
+            return Task.CompletedTask;
         }
```

Comment:
Remain other sample code?

Reply:
Only a POC for now, I want to make it clean...